### ISSUE TITLE
Expo web stuck in bundling loop

### ISSUE THREAD

**Issue by extralsc** [expo/expo#25508]
### Minimal reproducible example

https://github.com/extralsc/expo-router-web-bundling-issue.git

### Which package manager are you using? (Yarn is recommended)

npm

### If the issue is web-related, please select the bundler (`web.bundler` in the `app.json`)

metro

### Summary

Using the guide in expo for routes app creates app with a bug for web.
When starting for web the bundling is in a loop and redo it 3-4 times before completing the bundling.
It's easy to **reproduce** using: 
`npx create-expo-app@latest --template tabs@49`
then entering the folder for the installation and running `npx expo start --web`

Why does `npx expo start` and then running web create a endless loop of compiling for web?

I cant understand why this happens

This is what i see in terminal:

    Logs for your project will appear below. Press Ctrl+C to exit.
        Web node_modules/expo-router/entry.js ▓▓▓░░░░░░░░░░░░░ 24.4% ( 674/2454)
        Web node_modules/expo-router/node/render.js ▓▓▓▓░░░░░░░░░░░░ 25.8% ( 279/1995)
        Web node_modules/expo-router/entry.js ░░░░░░░░░░░░░░░░  0.0% (0/1)
        Web node_modules/expo-router/node/render.js ░░░░░░░░░░░░░░░░  0.0% (0/1)
        Web node_modules/expo-router/entry.js ░░░░░░░░░░░░░░░░  0.0% (0/1)
        Web node_modules/expo-router/node/render.js ░░░░░░░░░░░░░░░░  0.0% (0/1)
        Web node_modules/expo-router/entry.js ░░░░░░░░░░░░░░░░  0.0% (0/1)
        Web node_modules/expo-router/node/render.js ░░░░░░░░░░░░░░░░  0.0% (0/1)
        Web node_modules/expo-router/entry.js ░░░░░░░░░░░░░░░░  0.0% (0/1)
        Web node_modules/expo-router/node/render.js ░░░░░░░░░░░░░░░░  0.0% (0/1)

and after that:

    Web Bundling complete 125912ms
    Web Bundling complete 115721ms
    Web Bundling complete 115803ms
    Web Bundling complete 93894ms
    Web Bundling complete 93936ms
    Web Bundling complete 66106ms
    Web Bundling complete 127441ms
    Web Bundling complete 116727ms
    Web Bundling complete 116781ms
    Web Bundling complete 94849ms
    Web Bundling complete 94846ms
    Web Bundling complete 66980ms

then without using the web app automatically i get this rendered in console from time to time:

    Web Bundling complete 255ms
    Web Bundling complete 113ms
    "transform" style array value is deprecated. Use space-separated string functions, e.g., "scaleX(2) rotateX(15deg)".
    Web Bundling complete 233ms
    Web Bundling complete 104ms
    "transform" style array value is deprecated. Use space-separated string functions, e.g., "scaleX(2) rotateX(15deg)".
    Web Bundling complete 220ms
    Web Bundling complete 104ms
    "transform" style array value is deprecated. Use space-separated string functions, e.g., "scaleX(2) rotateX(15deg)".

Eventually it crashes and i get something like this in console:

    <--- Last few GCs --->
    
    [27462:0x6393420]   661440 ms: Mark-Compact (reduce) 2038.5 (2069.6) -> 2036.4 (2069.4) MB, 120.53 / 0.01 ms  (+ 1287.8 ms in 108 steps since start of marking, biggest step 54.0 ms, walltime since start of marking 1643 ms) (average mu = 0.798, current mu [27462:0x6393420]   662368 ms: Mark-Compact (reduce) 2037.3 (2069.4) -> 2036.4 (2070.4) MB, 914.42 / 0.00 ms  (average mu = 0.683, current mu = 0.014) allocation failure; scavenge might not succeed
    
    
    <--- JS stacktrace --->
    
    FATAL ERROR: Reached heap limit Allocation failed - JavaScript heap out of memory
     1: 0xc99970 node::Abort() [node]
     2: 0xb6ffcb  [node]
     3: 0xebe9f0 v8::Utils::ReportOOMFailure(v8::internal::Isolate*, char const*, v8::OOMDetails const&) [node]
     4: 0xebecd7 v8::internal::V8::FatalProcessOutOfMemory(v8::internal::Isolate*, char const*, v8::OOMDetails const&) [node]
     5: 0x10d0785  [node]
     6: 0x10e8608 v8::internal::Heap::CollectGarbage(v8::internal::AllocationSpace, v8::internal::GarbageCollectionReason, v8::GCCallbackFlags) [node]
     7: 0x10be721 v8::internal::HeapAllocator::AllocateRawWithLightRetrySlowPath(int, v8::internal::AllocationType, v8::internal::AllocationOrigin, v8::internal::AllocationAlignment) [node]
     8: 0x10bf8b5 v8::internal::HeapAllocator::AllocateRawWithRetryOrFailSlowPath(int, v8::internal::AllocationType, v8::internal::AllocationOrigin, v8::internal::AllocationAlignment) [node]
     9: 0x109bef6 v8::internal::Factory::AllocateRaw(int, v8::internal::AllocationType, v8::internal::AllocationAlignment) [node]
    10: 0x108d9f4 v8::internal::FactoryBase<v8::internal::Factory>::AllocateRawWithImmortalMap(int, v8::internal::AllocationType, v8::internal::Map, v8::internal::AllocationAlignment) [node]
    11: 0x1090da7 v8::internal::FactoryBase<v8::internal::Factory>::NewRawTwoByteString(int, v8::internal::AllocationType) [node]
    12: 0x154e28c v8::internal::IncrementalStringBuilder::Extend() [node]
    13: 0x120d7b1 v8::internal::JsonStringifier::SerializeString(v8::internal::Handle<v8::internal::String>) [node]
    14: 0x120ed51 v8::internal::JsonStringifier::Result v8::internal::JsonStringifier::Serialize_<true>(v8::internal::Handle<v8::internal::Object>, bool, v8::internal::Handle<v8::internal::Object>) [node]
    15: 0x1212ca9 v8::internal::JsonStringifier::Result v8::internal::JsonStringifier::Serialize_<false>(v8::internal::Handle<v8::internal::Object>, bool, v8::internal::Handle<v8::internal::Object>) [node]
    16: 0x1212237 v8::internal::JsonStringifier::Result v8::internal::JsonStringifier::Serialize_<false>(v8::internal::Handle<v8::internal::Object>, bool, v8::internal::Handle<v8::internal::Object>) [node]
    17: 0x1213a96 v8::internal::JsonStringify(v8::internal::Isolate*, v8::internal::Handle<v8::internal::Object>, v8::internal::Handle<v8::internal::Object>, v8::internal::Handle<v8::internal::Object>) [node]
    18: 0xf50ba2 v8::internal::Builtin_JsonStringify(int, unsigned long*, v8::internal::Isolate*) [node]
    19: 0x1931df6  [node]
    Aborted

**Expected behavior:**
Bundle once and does not get stuck in a loop.

**Actual behavior:**
Bundles several times, get stuck in a loop, working but eventually crashes.

### Environment

Node: v20.9.0
NPM: 10.1.0

Package.json:
```
{
  "name": "prod",
  "main": "expo-router/entry",
  "version": "1.0.0",
  "scripts": {
    "start": "expo start",
    "android": "expo start --android",
    "ios": "expo start --ios",
    "web": "expo start --web",
    "test": "jest --watchAll"
  },
  "jest": {
    "preset": "jest-expo"
  },
  "dependencies": {
    "@expo/vector-icons": "^13.0.0",
    "@react-navigation/native": "^6.0.2",
    "expo": "~49.0.18",
    "expo-font": "~11.4.0",
    "expo-linking": "~5.0.2",
    "expo-router": "^2.0.0",
    "expo-splash-screen": "~0.20.5",
    "expo-status-bar": "~1.6.0",
    "expo-system-ui": "~2.4.0",
    "expo-web-browser": "~12.3.2",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "react-native": "0.72.6",
    "react-native-gesture-handler": "~2.12.0",
    "react-native-safe-area-context": "4.6.3",
    "react-native-screens": "~3.22.0",
    "react-native-web": "~0.19.6"
  },
  "devDependencies": {
    "@babel/core": "^7.20.0",
    "@types/react": "~18.2.14",
    "jest": "^29.2.1",
    "jest-expo": "~49.0.0",
    "react-test-renderer": "18.2.0",
    "typescript": "^5.1.3"
  },
  "overrides": {
    "react-refresh": "~0.14.0"
  },
  "resolutions": {
    "react-refresh": "~0.14.0"
  },
  "private": true
}
```

**Comment by marklawlor** [comment]
I'm unable to replicate your issue, so it maybe environmental. Can you please reply with the output of running `npx expo-env-info`

**Comment by EvanBacon** [comment]
@extralsc could you please provide the steps of which files to modify to trigger the recursive loop? Also what browser/extensions are you using, perhaps a chrome extension is triggering updates?

**Comment by extralsc** [comment]
> I'm unable to replicate your issue, so it maybe environmental. Can you please reply with the output of running `npx expo-env-info`

Sure, here it is:
```
 expo-env-info 1.0.5 environment info:
    System:
      OS: Linux 5.15 Ubuntu 22.04.2 LTS 22.04.2 LTS (Jammy Jellyfish)
      Shell: 5.8.1 - /usr/bin/zsh
    Binaries:
      Node: 20.10.0 - ~/.nvm/versions/node/v20.10.0/bin/node
      Yarn: 1.22.21 - ~/.nvm/versions/node/v20.10.0/bin/yarn
      npm: 10.2.3 - ~/.nvm/versions/node/v20.10.0/bin/npm
    npmPackages:
      expo: ~49.0.18 => 49.0.20 
      react: 18.2.0 => 18.2.0 
      react-dom: 18.2.0 => 18.2.0 
      react-native: 0.72.6 => 0.72.6 
      react-native-web: ~0.19.6 => 0.19.9 
    npmGlobalPackages:
      expo-cli: 6.3.10
    Expo Workflow: managed
```

**Comment by extralsc** [comment]
> @extralsc could you please provide the steps of which files to modify to trigger the recursive loop? Also what browser/extensions are you using, perhaps a chrome extension is triggering updates?

All I did was to:
1. remove node_modules
2. npm i
3. npm start
4. w (To start web)
Leading to this screen:

```
Logs for your project will appear below. Press Ctrl+C to exit.
› Open in the web browser...
› Press ? │ show all commands
Web Bundling complete 9596ms
Web Bundling complete 9727ms
Web Bundling complete 9791ms
Web Bundling complete 9842ms
"transform" style array value is deprecated. Use space-separated string functions, e.g., "scaleX(2) rotateX(15deg)".
"transform" style array value is deprecated. Use space-separated string functions, e.g., "scaleX(2) rotateX(15deg)".
Web Bundling complete 125ms
Web Bundling complete 56ms
Web Bundling complete 80ms
Web Bundling complete 44ms
Web Bundling complete 82ms
Web Bundling complete 41ms
Web Bundling complete 80ms
Web Bundling complete 42ms
Web Bundling complete 93ms
Web Bundling complete 55ms
Web Bundling complete 164ms
Web Bundling complete 55ms
"transform" style array value is deprecated. Use space-separated string functions, e.g., "scaleX(2) rotateX(15deg)".
"transform" style array value is deprecated. Use space-separated string functions, e.g., "scaleX(2) rotateX(15deg)".
Web Bundling complete 45ms
Web node_modules/expo-router/entry.js ░░░░░░░░░░░░░░░░  0.0% (0/1)"transform" style array value is deprecated. Use space-separated string functions, e.g., "scaleX(2) rotateX(15deg)".
"transform" style array value is deprecated. Use space-separated string functions, e.g., "scaleX(2) rotateX(15deg)".
"transform" style array value is deprecated. Use space-separated string functions, e.g., "scaleX(2) rotateX(15deg)".
"transform" style array value is deprecated. Use space-separated string functions, e.g., "scaleX(2) rotateX(15deg)".
 LOG  [web] Logs will appear in the browser console
Web Bundling complete 148ms
Web Bundling complete 64ms
Web Bundling complete 99ms
Web Bundling complete 60ms
"transform" style array value is deprecated. Use space-separated string functions, e.g., "scaleX(2) rotateX(15deg)".
"transform" style array value is deprecated. Use space-separated string functions, e.g., "scaleX(2) rotateX(15deg)".
Web Bundling complete 98ms
Web Bundling complete 58ms
"transform" style array value is deprecated. Use space-separated string functions, e.g., "scaleX(2) rotateX(15deg)".
Web Bundling complete 122ms
Web Bundling complete 44ms
Web Bundling complete 153ms
Web Bundling complete 39ms
"transform" style array value is deprecated. Use space-separated string functions, e.g., "scaleX(2) rotateX(15deg)".
"transform" style array value is deprecated. Use space-separated string functions, e.g., "scaleX(2) rotateX(15deg)".
Web Bundling complete 83ms
Web Bundling complete 48ms
Web Bundling complete 78ms
Web Bundling complete 40ms
"transform" style array value is deprecated. Use space-separated string functions, e.g., "scaleX(2) rotateX(15deg)".
"transform" style array value is deprecated. Use space-separated string functions, e.g., "scaleX(2) rotateX(15deg)".
Web Bundling complete 91ms
Web Bundling complete 59ms
"transform" style array value is deprecated. Use space-separated string functions, e.g., "scaleX(2) rotateX(15deg)".
Web Bundling complete 80ms
Web Bundling complete 39ms
"transform" style array value is deprecated. Use space-separated string functions, e.g., "scaleX(2) rotateX(15deg)".
```
 
Note: This is a minimal example of a bigger project im using, i tried to remove as much code as possible without removing the problem itself.

At this point i have not even  edited any files.

Im using these chrome extensions:

Malwarebytes browser guard (that is turned off for localhost)
uBlock origin (Turned off for localhost)
AdBlock (Does not to seem be active, not finding any ads and blocking 0)
JSON Formatter
ExpressVPN (Turned off)
Tag Assistant Companion

**Comment by marklawlor** [comment]
I tested again an I'm unable to reproduce the issue. Env I tested with

```
  expo-env-info 1.0.5 environment info:
    System:
      OS: Linux 6.2 Ubuntu 22.04.3 LTS 22.04.3 LTS (Jammy Jellyfish)
      Shell: 5.8.1 - /usr/bin/zsh
    Binaries:
      Node: 20.10.0 - ~/.volta/tools/image/node/20.10.0/bin/node
      Yarn: 1.22.19 - ~/.volta/tools/image/yarn/1.22.19/bin/yarn
      npm: 10.2.3 - ~/.volta/tools/image/node/20.10.0/bin/npm
      Watchman: 4.9.0 - /usr/bin/watchman
    IDEs:
      Android Studio: AI-222.4459.24.2221.10121639
    npmPackages:
      expo: ~49.0.18 => 49.0.20 
      react: 18.2.0 => 18.2.0 
      react-dom: 18.2.0 => 18.2.0 
      react-native: 0.72.6 => 0.72.6 
      react-native-web: ~0.19.6 => 0.19.9 
    Expo Workflow: managed
```

Just to confirm, this always occurs with `npx create-expo-app@latest --template tabs@49` in a new project? 

Have you tried clearing the Metro cache with `npm start -- -c`?

Any there anything usual about your setup? Eg remote storage, symlinks, non A-Z characters for directory names?

**Comment by dealvz** [comment]
facing this exact issue, including the warning 
`"transform" style array value is deprecated. Use space-separated string functions, e.g., "scaleX(2) rotateX(15deg)".
`

expo-env-info yields
```expo-env-info 1.0.5 environment info:
    System:
      OS: macOS 14.1.1
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 16.19.0 - ~/.volta/tools/image/node/16.19.0/bin/node
      Yarn: 1.22.17 - ~/.volta/tools/image/yarn/1.22.17/bin/yarn
      npm: 8.19.3 - ~/.volta/tools/image/node/16.19.0/bin/npm
    Managers:
      CocoaPods: 1.12.0 - /usr/local/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 23.2, iOS 17.2, macOS 14.2, tvOS 17.2, watchOS 10.2
      Android SDK:
        API Levels: 33, 33
        Build Tools: 30.0.3, 33.0.0, 33.0.2, 34.0.0
        System Images: android-33 | Google APIs ARM 64 v8a
    IDEs:
      Android Studio: 2022.1 AI-221.6008.13.2211.9514443
      Xcode: 15.1/15C65 - /usr/bin/xcodebuild
    npmPackages:
      expo: ~49.0.11 => 49.0.21
      react: 18.2.0 => 18.2.0
      react-dom: 18.2.0 => 18.2.0
      react-native: 0.72.6 => 0.72.6
      react-native-web: ~0.19.6 => 0.19.9
    Expo Workflow: managed
    
    Has anyone found a solution?

**Comment by dealvz** [comment]
Okay it seems I found the issue
I was using Flipper to debug https://fbflipper.com/docs/features/react-native/
Closing this fixed the web bundling cycle and the "transform" warnings

**Comment by likeSo** [comment]
Same problem here after upgrade to SDK 50
```sh

Logs for your project will appear below. Press Ctrl+C to exit.
iOS Bundling complete 320251ms (node_modules/expo-router/entry.js)
iOS Bundling complete 1549ms (node_modules/expo-router/entry.js)
iOS Bundling complete 2642ms (node_modules/expo-router/entry.js)
iOS node_modules/expo-router/entry.js ░░░░░░░░░░░░░░░░  0.0% (0/1)
<--- Last few GCs --->

[36174:0x7fda7a84d000]   732073 ms: Scavenge 1835.8 (1884.4) -> 1829.5 (1884.4) MB, 8.0 / 0.0 ms  (average mu = 0.945, current mu = 0.954) allocation failure; 
[36174:0x7fda7a84d000]   733395 ms: Mark-sweep (reduce) 2055.2 (2103.2) -> 2040.3 (2087.8) MB, 672.6 / 0.0 ms  (+ 0.0 ms in 0 steps since start of marking, biggest step 0.0 ms, walltime since start of marking 678 ms) (average mu = 0.924, current mu = 0.57

<--- JS stacktrace --->

FATAL ERROR: Reached heap limit Allocation failed - JavaScript heap out of memory
 1: 0x10b279fa5 node::Abort() [/usr/local/bin/node]
 2: 0x10b27a195 node::OOMErrorHandler(char const*, bool) [/usr/local/bin/node]
 3: 0x10b3fc81c v8::internal::V8::FatalProcessOutOfMemory(v8::internal::Isolate*, char const*, bool) [/usr/local/bin/node]
 4: 0x10b5c10b5 v8::internal::Heap::FatalProcessOutOfMemory(char const*) [/usr/local/bin/node]
 5: 0x10b5bf950 v8::internal::Heap::CollectGarbage(v8::internal::AllocationSpace, v8::internal::GarbageCollectionReason, v8::GCCallbackFlags) [/usr/local/bin/node]
 6: 0x10b6578d1 v8::internal::ScavengeJob::Task::RunInternal() [/usr/local/bin/node]
 7: 0x10b2e5f1e node::PerIsolatePlatformData::RunForegroundTask(std::__1::unique_ptr<v8::Task, std::__1::default_delete<v8::Task>>) [/usr/local/bin/node]
 8: 0x10b2e4917 node::PerIsolatePlatformData::FlushForegroundTasksInternal() [/usr/local/bin/node]
 9: 0x10bd1046b uv__async_io [/usr/local/bin/node]
10: 0x10bd247ee uv__io_poll [/usr/local/bin/node]
11: 0x10bd109b8 uv_run [/usr/local/bin/node]
12: 0x10b1b4753 node::SpinEventLoop(node::Environment*) [/usr/local/bin/node]
13: 0x10b2be876 node::NodeMainInstance::Run() [/usr/local/bin/node]
14: 0x10b240a60 node::LoadSnapshotDataAndRun(node::SnapshotData const**, node::InitializationResult const*) [/usr/local/bin/node]
15: 0x10b240cce node::Start(int, char**) [/usr/local/bin/node]
16: 0x7ff819d7741f start [/usr/lib/dyld]
[1]    36159 abort      npx expo start -c
```

**Comment by imranbarbhuiya** [comment]
for me, it stuck initial bundling, after this, I cant event press `ctrl + c`

```sh
Starting project at D:\TEC\Spartans\apps\mobile-app
env: load .env.development.local
env: export EXPO_PUBLIC_API_BASE EXPO_PUBLIC_MESSAGE_BASE
Starting Metro Bundler

› Metro waiting on exp+spartans://expo-development-client/?url=............
› Scan the QR code above to open the project in a development build. Learn more: https://expo.fyi/start

› Web is waiting on http://localhost:8081

› Using development build
› Press s │ switch to Expo Go

› Press a │ open Android
› Press w │ open web

› Press j │ open debugger
› Press r │ reload app
› Press m │ toggle menu
› Press o │ open project code in your editor

› Press ? │ show all commands

Logs for your project will appear below. Press Ctrl+C to exit.
› Open in the web browser...
› Press ? │ show all commands
Web D:\TEC\Spartans\apps\mobile-app/index.ts ░░░░░░░░░░░░░░░░  0.0% (0/1)done
Web D:\TEC\Spartans\apps\mobile-app/index.ts ▓▓▓▓▓▓▓▓▓░░░░░░░ 60.8% (756/986)
```

**Comment by raulghm** [comment]
> Same problem here after upgrade to SDK 50
> 
> ```shell
> Logs for your project will appear below. Press Ctrl+C to exit.
> iOS Bundling complete 320251ms (node_modules/expo-router/entry.js)
> iOS Bundling complete 1549ms (node_modules/expo-router/entry.js)
> iOS Bundling complete 2642ms (node_modules/expo-router/entry.js)
> iOS node_modules/expo-router/entry.js ░░░░░░░░░░░░░░░░  0.0% (0/1)
> <--- Last few GCs --->
> 
> [36174:0x7fda7a84d000]   732073 ms: Scavenge 1835.8 (1884.4) -> 1829.5 (1884.4) MB, 8.0 / 0.0 ms  (average mu = 0.945, current mu = 0.954) allocation failure; 
> [36174:0x7fda7a84d000]   733395 ms: Mark-sweep (reduce) 2055.2 (2103.2) -> 2040.3 (2087.8) MB, 672.6 / 0.0 ms  (+ 0.0 ms in 0 steps since start of marking, biggest step 0.0 ms, walltime since start of marking 678 ms) (average mu = 0.924, current mu = 0.57
> 
> <--- JS stacktrace --->
> 
> FATAL ERROR: Reached heap limit Allocation failed - JavaScript heap out of memory
>  1: 0x10b279fa5 node::Abort() [/usr/local/bin/node]
>  2: 0x10b27a195 node::OOMErrorHandler(char const*, bool) [/usr/local/bin/node]
>  3: 0x10b3fc81c v8::internal::V8::FatalProcessOutOfMemory(v8::internal::Isolate*, char const*, bool) [/usr/local/bin/node]
>  4: 0x10b5c10b5 v8::internal::Heap::FatalProcessOutOfMemory(char const*) [/usr/local/bin/node]
>  5: 0x10b5bf950 v8::internal::Heap::CollectGarbage(v8::internal::AllocationSpace, v8::internal::GarbageCollectionReason, v8::GCCallbackFlags) [/usr/local/bin/node]
>  6: 0x10b6578d1 v8::internal::ScavengeJob::Task::RunInternal() [/usr/local/bin/node]
>  7: 0x10b2e5f1e node::PerIsolatePlatformData::RunForegroundTask(std::__1::unique_ptr<v8::Task, std::__1::default_delete<v8::Task>>) [/usr/local/bin/node]
>  8: 0x10b2e4917 node::PerIsolatePlatformData::FlushForegroundTasksInternal() [/usr/local/bin/node]
>  9: 0x10bd1046b uv__async_io [/usr/local/bin/node]
> 10: 0x10bd247ee uv__io_poll [/usr/local/bin/node]
> 11: 0x10bd109b8 uv_run [/usr/local/bin/node]
> 12: 0x10b1b4753 node::SpinEventLoop(node::Environment*) [/usr/local/bin/node]
> 13: 0x10b2be876 node::NodeMainInstance::Run() [/usr/local/bin/node]
> 14: 0x10b240a60 node::LoadSnapshotDataAndRun(node::SnapshotData const**, node::InitializationResult const*) [/usr/local/bin/node]
> 15: 0x10b240cce node::Start(int, char**) [/usr/local/bin/node]
> 16: 0x7ff819d7741f start [/usr/lib/dyld]
> [1]    36159 abort      npx expo start -c
> ```

same here, anyone how fix that?

**Comment by Jianlong-Nie** [comment]
Same issue when i run ios
![image](https://github.com/expo/expo/assets/5245704/f7280251-f320-4bda-b137-dedeb8dd4b2d)

**Comment by marklawlor** [comment]
I'm going to close this issue as it appears that the originally reported is of **Expo Web** infinitely bundling was identified as a Flipper issue.

@raulghm @Jianlong-Nie Please create new issue for your Expo iOS issues. Please include a reproduction, as we cannot debug screenshots or code snippets.

**Comment by aflatoon2874** [comment]
Hello,

I also faced the same issue of recursive build for web and ultimately metro crashing due to out of memory error. The problem in my case was that I had set `expo.web.output` to `static`. I fixed it by setting the value to `single` for SPA.

**Comment by treasure-praise** [comment]
> Okay it seems I found the issue I was using Flipper to debug https://fbflipper.com/docs/features/react-native/ Closing this fixed the web bundling cycle and the "transform" warnings

this solved the issue for me thanks @dealvz

**Comment by luffy-taro** [comment]
For me it was an issue with nativewind 4 on windows

See https://github.com/nativewind/nativewind/issues/610

**Comment by alitnk** [comment]
Having the same issue here. I confirm removing `"output": "static"` "fixes" it but I still would like to use the `static` output option if possible. I'm not using Flipper (not that I'm aware of).

**Comment by patrickkeenan** [comment]
I'm getting this same issue, it started happening after I changed output to "server" from "static", but now when I switch it back, I still have the reloading loop.

Here's my app.json
```

{
  "expo": {
...
    "assetBundlePatterns": [
      "**/*"
    ],
    "newArchEnabled": true,
...
    "web": {
      "bundler": "metro",
      "favicon": "./assets/images/icon.png",
      "output": "server",
      "build": {
        "babel": {
          "include": [
            "@expo/vector-icons"
          ]
        }
      }
    },
    "plugins": [
      "expo-router",
      [
        "expo-build-properties",
        {
          "android": {
            "compileSdkVersion": 34,
            "targetSdkVersion": 34,
            "minSdkVersion": 34,
            "buildToolsVersion": "34.0.0"
          }
        }
      ],
      "expo-secure-store",
      [
        "expo-font",
        {
          "fonts": []
        }
      ],
      [
        "expo-splash-screen",
        {
          "image": "./assets/images/splash-icon.png",
          "imageWidth": 200,
          "resizeMode": "contain",
          "backgroundColor": "#000000"
        }
      ],
      "expo-secure-store",
      [
        "expo-location",
        {
          "locationAlwaysAndWhenInUsePermission": "Allow .... to use your location to show your position on the map and validate quest locations."
        }
      ]
    ],
    "experiments": {
      "typedRoutes": true,
      "tsconfigPaths": true
    },
    "extra": {
      "googleMapsApiKey": "...",
      "router": {
        "origin": false
      },
      "eas": {
        "projectId": "..."
      }
    },
    "runtimeVersion": {
      "policy": "appVersion"
    },
    "updates": {
      "url": "...."
    }
  }
}
```

**Comment by JamieGoodson** [comment]
I am also stuck in a bundling loop. This is very frustrating.

```
...
λ Bundled 12ms node_modules/expo-router/node/render.js (1 module)
Web Bundled 77ms node_modules/.pnpm/expo-router@5.0.3_@types+react@19.1.2_expo-constants@17.1.4_expo@53.0.4_@babel+core@7.26.9_@e_wmdp7rdd6vnjoybjyt5s5skcwm/node_modules/expo-router/entry.js (1 module)
Web node_modules/.pnpm/expo-router@5.0.3_@types+react@19.1.2_expo-constants@17.1.4_expo@53.0.4_@babel+core
λ Bundled 18ms node_modules/expo-router/node/render.js (1 module)
Web Bundled 91ms node_modules/.pnpm/expo-router@5.0.3_@types+react@19.1.2_expo-constants@17.1.4_expo@53.0.4_@babel+core@7.26.9_@e_wmdp7rdd6vnjoybjyt5s5skcwm/node_modules/expo-router/entry.js (1 module)
Web node_modules/.pnpm/expo-router@5.0.3_@types+react@19.1.2_expo-constants@17.1.4_expo@53.0.4_@babel+core
λ Bundled 18ms node_modules/expo-router/node/render.js (1 module)
Web Bundled 81ms node_modules/.pnpm/expo-router@5.0.3_@types+react@19.1.2_expo-constants@17.1.4_expo@53.0.4_@babel+core@7.26.9_@e_wmdp7rdd6vnjoybjyt5s5skcwm/node_modules/expo-router/entry.js (1 module)
...
```

(Repeats indefinitely)

**Comment by alitsvin** [comment]
I changed `expo.web.output` from "static" to "standalone", and it built successfully. Then just changed it back.
